### PR TITLE
Add section about Constrained classes to the Migration Guide

### PR DESCRIPTION
**Pull request by Kludex**
Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by Kludex** [comment]
please review

File: docs/migration.md [hunk 1]
```diff
@@ -749,6 +749,36 @@ classes using `Annotated`.
...
+```
+
+...becomes:
+
+```py
+from pydantic import BaseModel, Field
+
+
+class Model(BaseModel):
+    x: Annotated[int, Field(ge=0)]
```
Comment (adriangb) [inline docs/migration.md:775]: I’d make it a type alias so that it’s closer to the original example

**Review by adriangb — APPROVED** [review]

File: docs/migration.md (hunk 1 above)
Comment (Kludex) [inline docs/migration.md:775]:
```suggestion

MyInt = Annotated[int, Field(ge=0)]

class Model(BaseModel):
    x: MyInt
```

File: docs/migration.md [hunk 2]
```diff
@@ -749,6 +749,38 @@ classes using `Annotated`.
...
+```
+
+...becomes:
+
+```py
+from pydantic import BaseModel, Field
+
+
+MyInt = Annotated[int, Field(ge=0)]
+
```
Comment (Kludex) [inline docs/migration.md:775]:
```suggestion


```

File: docs/migration.md [hunk 3]
```diff
@@ -749,6 +749,39 @@ classes using `Annotated`.
...
+
+class Model(BaseModel):
+    x: MyInt
+```
+
+...becomes:
+
+```py
+from pydantic import BaseModel, Field
```
Comment (Kludex) [inline docs/migration.md:771]:
```suggestion
from typing_extensions import Annotated

from pydantic import BaseModel, Field
```

**Pull request merged**